intake_esm/cesm.py

```diff
@@ -342,51 +327,70 @@ def results(self):
                 self.collection_name,
                 self.collection_type,
                 self.query,
-                order_by=['sequence_order', 'files'],
+                order_by=['sequence_order', 'file_fullpath'],
             )
             return self.query_results
 
+    def to_xarray(self, **kwargs):
+        """Return dataset as an xarray dataset
+
+        Parameters
+        ----------
+        ** kwargs :
+               Additional keyword arguments are passed through to methods in aggregate.py
+        """
+        _kwargs = self.kwargs.copy()
+        _kwargs.update(kwargs)
+        self.kwargs = _kwargs
+        return self.to_dask()
+
     def _open_dataset(self):
-        url = self.urlpath
-        kwargs = self._kwargs
+        kwargs = self.kwargs
 
-        if len(self.query_results) == 0:
-            raise ValueError('query results are empty')
+        if self.query_results.empty:
+            raise ValueError(f'Query={self.query} returned empty results')
 
         query = dict(self.query)
-        if '*' in url or isinstance(url, list):
-            if 'concat_dim' not in kwargs.keys():
-                kwargs.update(concat_dim=self.concat_dim)
-            if self.pattern:
-                kwargs.update(preprocess=self._add_path_to_ds)
-
-            ensembles = self.query_results.ensemble.unique()
-            variables = self.query_results.variable.unique()
-
-            ds_ens_list = []
-            for ens_i in ensembles:
-                query['ensemble'] = ens_i
-
-                ds_var_list = []
-                for var_i in variables:
-
-                    query['variable'] = var_i
-                    urlpath_ei_vi = get_subset(
-                        self.collection_name,
-                        self.collection_type,
-                        query,
-                        order_by=['sequence_order', 'files'],
-                    ).files.tolist()
-
-                    dsets = [
-                        aggregate.open_dataset(url, data_vars=[var_i]) for url in urlpath_ei_vi
-                    ]
-                    ds_var_i = aggregate.concat_time_levels(dsets)
-                    ds_var_list.append(ds_var_i)
-
-                ds_ens_i = aggregate.merge(ds_var_list)
-                ds_ens_list.append(ds_ens_i)
-
-            self._ds = aggregate.concat_ensembles(ds_ens_list, member_ids=ensembles, join='outer')
-        else:
-            self._ds = xr.open_dataset(url, chunks=self.chunks, **kwargs)
+
+        if 'decode_times' not in kwargs.keys():
+            kwargs.update(decode_times=False)
+        if 'time_coord_name' not in kwargs.keys():
+            kwargs.update(time_coord_name='time')
```

Comment:
Should this be a call to `infer_time_coord_name`?

Reply:
At this stage, we don't have access to any xarray dataset yet. So, `infer_time_coord_name()` wouldn't work. Even though the default value here is `time`, in the next steps, I call `infer_time_coord_name()` to confirm that this is the right time coordinate name prior doing any operation

Reply:
Fair enough. Should `infer_time_coord_name` be renamed to `ensure_time_coord`?

Reply:
Done!